### ISSUE TITLE
Allow adding priorities to tasks

### ISSUE THREAD

**Issue by robinchrist** [taskflow/taskflow#232]
I'm kind of abusing taskflow as a work scheduler / load balancer.

It would be very useful if I could add priorities to tasks or at least individual graphs...
Do you reckon this is possible? I believe that would at least require some changes to the workstealing queue to work as a lockless priority workstealing queue but I don't know enough about the source code to give a more precise estimate.

**Comment by tsung-wei-huang** [comment]
@robinchrist It depends on how may priority levels you need. Typically, we can have one work-stealing queue per priority level, of course, at the cost of extra memory. That being said, the scheduler will not guarantee everything will go through the priority. I will leave this as an enhancement to discuss but definitely feel free to contribute.

**Comment by dasmig** [comment]
Well, I just wanted to endorse this request allowing adding priorities to task. If we could have at least one priority queue it would greatly improve our usage of the library. In our use case, we allocate a lot of resources through our program, so not being able to prioritize the later tasks which make use of what we allocated and deallocate it is very frustrating. I took a look at the library source code to try adding it myself but I didn't feel confident to do it. I'm not sure how hard it would be nor if it would be useful to many other people.
